service/src/main/java/uk/nhs/adaptors/gp2gp/common/storage/StorageConnectorService.java

```diff
@@ -0,0 +1,36 @@
+package uk.nhs.adaptors.gp2gp.common.storage;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import static uk.nhs.adaptors.gp2gp.gpc.GpcFileNameConstants.GPC_STRUCTURED_FILE_EXTENSION;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.AllArgsConstructor;
+import uk.nhs.adaptors.gp2gp.gpc.GpcStructuredResponseObject;
+
+@Service
+@AllArgsConstructor
+public class StorageConnectorService {
+    private final StorageConnectorFactory storageConnectorFactory;
+    private final ObjectMapper objectMapper;
+
+    @SuppressFBWarnings(
```

Comment:
Doh! I totally missed the actual issue here.

Don't ever use `StorageConnectorFactory` directly. This is something Spring uses behind-the-scenes to create an instance of StorageConnector.

Instead just declare `private final StorageConnector storageConnector;` and let spring inject the dependency.